### PR TITLE
draft: --add-timestamp

### PR DESCRIPTION
**Pull request by omry**
introducing an --add_timestamp flag and support for in in the logging system.

#289

### CODE REVIEW

**Comment by omry** [comment]
1. Looking for overall feedback.
2. colorlog test is failing even though things are working. looks like some kind of interaction between colorlog and logcap.

**Comment by omry** [comment]
Screen shots:

image
image

**Comment by omry** [comment]
@theacodes ping

File: nox/_options.py [hunk 1]
```diff
@@ -221,6 +221,15 @@ def _session_completer(
         help="Logs the output of all commands run including commands marked silent.",
         noxfile=True,
     ),
+    _option_set.Option(
+        "add_timestamp",
+        "-ts",
+        "--add_timestamp",
```
Comment (theacodes) [inline nox/_options.py:227]: should be `--add-timestamp`.

**Comment by theacodes** [comment]
> Looking for overall feedback (Is it okay to add the flag? is the log format okay? [I noticed it's missing nox> )

This is a great idea. I would definitely add the `nox >`, as it's important to be able to distinguish Nox's output from the output that programs called via `run` produce.

> colorlog test is failing even though things are working. looks like some kind of interaction between colorlog and logcap.

We'll definitely wanna get that working, but I'm not a fan of using subprocess in tests unless its absolutely necessary.

**Comment by omry** [comment]
> We'll definitely wanna get that working, but I'm not a fan of using subprocess in tests unless its absolutely necessary.

The problem I ran into was that for an unknown reason, asctime was not populated in the caplog record when using colorlog. but we are talking about an interaction between two third party libs (caplog and colorlib), through the somewhat complicated Python logging subsystem.

I already spent about an hour or two digging into it. I will give it another shot before resorting to process testing.

**Comment by omry** [comment]
responded to comments, except the test.
Upon further digging, I suspect custom log formatter are [not fully supported](https://github.com/pytest-dev/pytest/issues/2987) by pytest caplog.

I see two options:
1. Drop the testing timestamp override testing for the colorlog. This is very visible so if it breaks we will hear about it and keep the current working testing for non-colorlog mode.
2. Switch both test modes to process based testing.

What do you think?

**Comment by theacodes** [comment]
Sorry for taking so long to get to this. Let's go with (1).

**Comment by omry** [comment]
### Regular run
image

### Run with timestamps
image

### Runt with timestamp and no color.
image

**Comment by omry** [comment]
I noticed that there is some cleanup to do:
1. blacken is not running and seems to be producing results different than `black .`. it should probably be removed from the noxfile.
2. docs/conf.py needs black formatting.

**Comment by theacodes** [comment]
Thanks!

**Pull request merged**